test/files/neg/patmat-reachable-partial.scala

```diff
@@ -0,0 +1,6 @@
+// scalac: -Xfatal-warnings
+
+trait C
+class Test {
+  def test(c: Seq[Option[Int]]) = c.collect { case _: C => }
+}
```

Comment:
Worth noting that a different analysis in typer would report a "fruitless type test" error if `C` is a class and the scrutinee type (here, `Option`) was not in the same compilation run as `def test`.

```
scala> class C; def test(c: Seq[Option[Int]]) = c.collect { case _: C => }
<console>:12: warning: fruitless type test: a value of type Option[Int] cannot also be a C
       class C; def test(c: Seq[Option[Int]]) = c.collect { case _: C => }
                                                                    ^
defined class C
test: (c: Seq[Option[Int]])Seq[Unit]
```

```scala
    /** Is it impossible for the given symbols to be parents in the same class?
     *  This means given A and B, can there be an instance of A with B? This is the
     *  case if neither A nor B is a subclass of the other, and one of the following
     *  additional conditions holds:
     *   - either A or B is effectively final
     *   - neither A nor B is a trait (i.e. both are actual classes, not eligible for mixin)
     *   - both A and B are sealed/final, and every possible pairing of their children is irreconcilable
     *
     *  TODO: the last two conditions of the last possibility (that the symbols are not of
     *  classes being compiled in the current run) are because this currently runs too early,
     *  and .children returns Nil for sealed classes because their children will not be
     *  populated until typer.  It was too difficult to move things around for the moment,
     *  so I will consult with moors about the optimal time to be doing this.
     */
    def areIrreconcilableAsParents(sym1: Symbol, sym2: Symbol): Boolean = areUnrelatedClasses(sym1, sym2) && (
         isEffectivelyFinal(sym1) // initialization important
      || isEffectivelyFinal(sym2)
      || !sym1.isTrait && !sym2.isTrait
      || isSealedOrFinal(sym1) && isSealedOrFinal(sym2) && allChildrenAreIrreconcilable(sym1, sym2) && !currentRun.compiles(sym1) && !currentRun.compiles(sym2)
    )

```